dashboard/app/controllers/featured_projects_controller.rb

```diff
@@ -0,0 +1,19 @@
+class FeaturedProjectsController < ApplicationController
+  def create
+    storage_app_id = storage_decrypt_channel_id(featured_project_params[:project_id])
+    @featured_project = FeaturedProject.create({storage_app_id: storage_app_id[0], who_featured_user_id: current_user.id})
```

Comment:
Is `storage_app_id[0]` the best way to get the id for the project? I'm still unclear about the differences between storage_id and channel_id, which seemed from our conversation to both be synonymous with project id? 
 https://github.com/code-dot-org/code-dot-org/blob/2092395748627d3b55a9cb34aa85ba0ef43779fa/shared/middleware/helpers/storage_id.rb#L46

Reply:
Hi Erin, unfortunately this is a part of the system where the terminology is a bit of a mess. Here is how things currently stand:

terminology:
* storage_id: this is an ID representing a user. it does not represent a project and is not the same as a channel_id. for signed-in users, storage_id is linked to a exactly one user_id via the user_storage_ids table. for signed-out users, the storage_id is stored directly in a cookie.
* channel_id: this is the id for an entry in the storage_apps table, which represents a project. these are also sometimes called storage_app_id when we are talking about them at the database level.
* encrypted_channel_id: this is an encrypted combination of storage_id and channel_id, which you get by calling storage_encrypt_channel_id(). This is, unfortunately, also sometimes called a channel_id. It is also sometimes called a project_id.

the best way to get the channel_id is `_, channel_id = storage_decrypt_channel_id(...)`. Or, if you need the storage_id too, you can do `storage_id, channel_id = storage_decrypt_channel_id(...)`